xlml/utils/metric.py

```diff
@@ -332,6 +376,50 @@ def add_airflow_metadata(
   return metadata
 
 
+def add_test_config_metadata(
+    base_id: str,
+    task_test_config: test_config.TestConfig[test_config.Tpu],
+    task_gcp_config: gcp_config.GCPConfig,
+    metadata: List[List[bigquery.MetricHistoryRow]],
+) -> List[List[bigquery.MetricHistoryRow]]:
+  for index in range(len(metadata)):
+    uuid = generate_row_uuid(base_id, index)
+    test_config_meta = []
+
+    test_config_meta.append(
+        bigquery.MetadataHistoryRow(
+            job_uuid=uuid,
+            metadata_key="accelerator",
+            metadata_value=task_test_config.accelerator.name,
+        )
+    )
+    test_config_meta.append(
+        bigquery.MetadataHistoryRow(
+            job_uuid=uuid,
+            metadata_key="project",
+            metadata_value=task_gcp_config.project_name,
+        )
+    )
+    if hasattr(task_test_config, "num_slices"):
```

Comment:
Will you meet error without this? I see `TestConfig[Tpu]` is defined here

Reply:
I changed the parameter to be `TestConfig[test_config.Accelerator]` since this function would also be called for GPU tests. I think we need the if since there is no `num_slices` in `GpuVmTest` https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/5e0d91f69f6a68640d3d8be6ffa486e961d3fd7b/xlml/apis/test_config.py#L188

Reply:
SG! Thanks!